tests/test_util.py

```diff
@@ -48,3 +49,34 @@ def test_bidict(self):
         bd.clear()
         assert bd == {}
         assert bd.inverse == {}
+
+
+class TestTemplateFilters:
+    def test_calc_memory(self):
+        class MockOp:
+            def __init__(self, memory=None):
+                self.directives = {"memory": memory}
+
+        op1 = MockOp(1)
+        op2 = MockOp(8)
+        op3 = MockOp()
```

Comment:
You could define these as fixtures then reuse them in `test_check_memory` below.

Reply:
Test check_memory is no longer needed but I'll still define them as fixtures.